src/xradio/image/_util/_zarr/zarr_low_level.py

```diff
@@ -311,36 +340,58 @@ def create_data_variable_meta_data_on_disk(
         zarr_meta[data_variable_key]["chunks"] = chunks
         zarr_meta[data_variable_key]["shape"] = shape
 
-        write_json_file(zarray, os.path.join(data_variable_path, ".zarray"))
+        # again, assuming data_variable_path has been set compatibly
+        zarray_file = os.path.join(data_variable_path, ".zarray")
+
+        if isinstance(fs, s3fs.core.S3FileSystem):
+            with fs.open(zattrs_file, "w") as file:
+                json.dump(
+                    zattrs,
+                    file,
+                    indent=4,
+                    sort_keys=True,
+                    ensure_ascii=True,
+                    separators=(",", ": "),
+                    cls=NumberEncoder,
+                )
+        else:
+            # default to assuming we can use primitives
+            write_json_file(zarray, zarray_file)
+
     return zarr_meta
 
 
 def write_chunk(img_xds, meta, parallel_dims_chunk_id, compressor, image_file):
     dims = meta["dims"]
     dtype = meta["dtype"]
-    data_varaible_name = meta["name"]
+    data_variable_name = meta["name"]
     chunks = meta["chunks"]
     shape = meta["shape"]
     chunk_name = ""
-    if data_varaible_name in img_xds:
-        for d in img_xds[data_varaible_name].dims:
+    if data_variable_name in img_xds:
+        for d in img_xds[data_variable_name].dims:
             if d in parallel_dims_chunk_id:
                 chunk_name = chunk_name + str(parallel_dims_chunk_id[d]) + "."
             else:
                 chunk_name = chunk_name + "0."
         chunk_name = chunk_name[:-1]
 
-        if list(img_xds[data_varaible_name].shape) != list(chunks):
+        if list(img_xds[data_variable_name].shape) != list(chunks):
             array = pad_array_with_nans(
-                img_xds[data_varaible_name].values,
+                img_xds[data_variable_name].values,
                 output_shape=chunks,
                 dtype=dtype,
             )
         else:
-            array = img_xds[data_varaible_name].values
-
-        write_binary_blob_to_disk(
-            array,
-            file_path=os.path.join(image_file, data_varaible_name, chunk_name),
+            array = img_xds[data_variable_name].values
+
+        z_chunk = zarr.open(
+            os.path.join(image_file, data_variable_name, chunk_name),
+            mode="a",
+            shape=meta["shape"],
+            chunks=meta["chunks"],
+            dtype=meta["dtype"],
```

Comment:
This is the basic change requested by https://github.com/casangi/astroviper/issues/18. It might have been possible to extend `write_binary_blob_to_disk` to handle S3 URIs but just using `zarr.open` directly makes sense to me.